apps/emqx_ds_builtin_local/src/emqx_ds_builtin_local.erl

```diff
@@ -380,3 +397,20 @@ timeus_to_timestamp(undefined) ->
     undefined;
 timeus_to_timestamp(TimestampUs) ->
     TimestampUs div 1000.
+
+with_worker(F, A) ->
+    Parent = self(),
+    Ref = make_ref(),
+    {_Pid, MRef} = spawn_opt(
+        fun() ->
+            Parent ! {Ref, apply(?MODULE, F, A)}
+        end,
+        [monitor, {min_heap_size, 10000}]
+    ),
```

Comment:
I don't think this needs a lot of testing because Raft backend always runs table scan in a worker process, even for the calls that end up at the local node. (There's a gen_rpc optimization that bypasses network layer for such calls, but it still spawns a temporary process for various reasons). Any code that uses an g?e(n_)?rpc implicitly does the same thing. 
And my tests show that inhibiting GC induced by running table scan in the same process leads to better throughput.

Reply:
> And my tests show that inhibiting GC induced by running table scan in the same process leads to more stable throughput.

Even if, say, there are only 2 new messages in a stream?

Reply:
Like I said, the "optimization" of running scan in the same process is not going to stick, whether it's more or not.

1) It's specific to builtin local backend (which is not our top priority). If something can't be reused in Raft (or other HA backend), we don't implement it in local.
2) It can't be used with async fetch (must have for *session* performance), we don't want to block PUBACKS and mess up MQTT channel flow control